src/Features/Core/Portable/CodeFixes/Suppression/ConfigureSeverityLevelCodeAction.cs

```diff
@@ -0,0 +1,283 @@
+                        var lastLine = lines.AsEnumerable().LastOrDefault();
+                        if (lastLine.ToString().Trim().Length != 0)
+                        {
+                            newRule = "\r\n";
+                        }
+                        // Insert newline if file is not empty
+                        if (lines.AsEnumerable().Count() > 1 && lastLine.ToString().Trim().Length == 0)
+                        {
+                            newRule += "\r\n";
+                        }
+
+                        if (language == LanguageNames.CSharp)
+                        {
+                            newRule += "[*.cs]\r\n" + name + " = " + option + ":" + severity + "\r\n";
+                        }
+                        else if (language == LanguageNames.VisualBasic)
+                        {
+                            newRule += "[*.vb]\r\n" + name + " = " + option + ":" + severity + "\r\n";
+                        }
+                        var textChange = new TextChange(new TextSpan(result.Length, 0), newRule);
+                        solution = solution.WithAdditionalDocumentText(editorconfig.Id, result.WithChanges(textChange));
+                    }
+                }
+            }
+        }
+
+        internal static readonly Dictionary<string, PerLanguageOption<CodeStyleOption<bool>>> diagnosticToEditorConfigDotNet = new Dictionary<string, Options.PerLanguageOption<CodeStyleOption<bool>>>()
+        {
+            { IDEDiagnosticIds.UseThrowExpressionDiagnosticId, CodeStyleOptions.PreferThrowExpression },
+            { IDEDiagnosticIds.UseObjectInitializerDiagnosticId, CodeStyleOptions.PreferObjectInitializer },
+            { IDEDiagnosticIds.InlineDeclarationDiagnosticId, CodeStyleOptions.PreferInlinedVariableDeclaration },
+            { IDEDiagnosticIds.UseCollectionInitializerDiagnosticId, CodeStyleOptions.PreferCollectionInitializer },
+            { IDEDiagnosticIds.UseCoalesceExpressionDiagnosticId, CodeStyleOptions.PreferCoalesceExpression },
+            { IDEDiagnosticIds.UseCoalesceExpressionForNullableDiagnosticId, CodeStyleOptions.PreferCoalesceExpression },
+            { IDEDiagnosticIds.UseNullPropagationDiagnosticId, CodeStyleOptions.PreferNullPropagation },
+            { IDEDiagnosticIds.UseAutoPropertyDiagnosticId, CodeStyleOptions.PreferAutoProperties },
+            { IDEDiagnosticIds.UseExplicitTupleNameDiagnosticId, CodeStyleOptions.PreferExplicitTupleNames },
+            { IDEDiagnosticIds.UseDeconstructionDiagnosticId, CodeStyleOptions.PreferDeconstructedVariableDeclaration },
+            { IDEDiagnosticIds.MakeFieldReadonlyDiagnosticId, CodeStyleOptions.PreferReadonly },
+            { IDEDiagnosticIds.UseConditionalExpressionForAssignmentDiagnosticId, CodeStyleOptions.PreferConditionalExpressionOverAssignment },
+            { IDEDiagnosticIds.UseConditionalExpressionForReturnDiagnosticId, CodeStyleOptions.PreferConditionalExpressionOverReturn }
```

Comment:
More tight coupling.  How would anyone know they would ever need to update these areas of the code when changing any sort of feature/analyzer/fixer/etc.?